### PR TITLE
Wait and assert no running instances in tests and prefix test instances [HZ-1842]

### PR DESCRIPTION
**Pull request by ldziedziul**
Maybe fixes: https://github.com/hazelcast/hazelcast-enterprise/issues/5519
This PR:
- Adds waiting for asynchronous shutdown of the test instances up to 30 seconds
- Adds test class name as a prefix for the created instances to easily see find the creator of the instance
```diff
- 13:36:47,234  INFO || - [ClusterService] hz.happy_fermi.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] 
+ 13:36:47,234  INFO || - [ClusterService] hz.TSSqlIndexTest_happy_fermi.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] 
```

Wee see failures of `SimpleTestInClusterSupport#assertNoRunningInstances` in `@BeforeClass` but it's not obvious where from we got these leftover instances.
```
[There should be no running instances] 
Expecting empty but was:<[HazelcastInstance{name='stoic_northcutt', node=[127.0.0.1]:5701}]>
```

BTW It's odd since in `com.hazelcast.test.AfterClassesStatement` we terminate all instances

### CODE REVIEW

**Comment by frant-hartm** [comment]
run-ee-tests

**Review by TomaszGaweda — APPROVED** [review]

**Review by mdumandag — APPROVED** [review]

File: hazelcast/src/test/java/com/hazelcast/jet/SimpleTestInClusterSupport.java [hunk 1]
```diff
@@ -60,7 +59,7 @@ public abstract class SimpleTestInClusterSupport extends JetTestSupport {
     private static HazelcastInstance client;
 
     protected static void initialize(int memberCount, @Nullable Config config) {
-        assertNoRunningInstances();
+        assertTrueEventually(HazelcastTestSupport::assertNoRunningInstances, 30);
```
Comment (frant-hartm) [inline hazelcast/src/test/java/com/hazelcast/jet/SimpleTestInClusterSupport.java:62]: I think this should be reverted back.

File: hazelcast/src/test/java/com/hazelcast/test/AfterClassesStatement.java [hunk 2]
```diff
@@ -44,6 +45,7 @@ public void evaluate() throws Throwable {
         if (!instances.isEmpty()) {
             String message = "Instances haven't been shut down: " + instances;
             HazelcastInstanceFactory.terminateAll();
+            assertTrueEventually(HazelcastTestSupport::assertNoRunningInstances, 30);
```
Comment (frant-hartm) [inline hazelcast/src/test/java/com/hazelcast/test/AfterClassesStatement.java:48]: :+1: 

We could do the same for the clients below.

**Review by frant-hartm — CHANGES_REQUESTED** [review]

File: hazelcast/src/test/java/com/hazelcast/test/AfterClassesStatement.java (hunk 2 above)
Comment (ldziedziul) [reply hazelcast/src/test/java/com/hazelcast/test/AfterClassesStatement.java:48]: Added: https://github.com/hazelcast/hazelcast/pull/23043/commits/784628f8b217c9e55a0f96c5650ea41b2a878782

File: hazelcast/src/main/java/com/hazelcast/instance/impl/MobyNames.java [hunk 3]
```diff
@@ -23,7 +23,8 @@
 /**
  * Java port of the Moby Project random name generator (https://github.com/moby/moby).
  */
-final class MobyNames {
+public final class MobyNames {
+    public static final String MOBY_NAMING_PREFIX = "hazelcast.member.naming.moby.prefix";
```
[the remark this answers is not in the document]
Comment (ldziedziul) [reply hazelcast/src/main/java/com/hazelcast/instance/impl/MobyNames.java]: Done: https://github.com/hazelcast/hazelcast/pull/23043/commits/b46c27d88bbdd760fe824ef4258aed33990e8fee

**Review by frant-hartm — APPROVED** [review]

**Comment by ldziedziul** [comment]
run-lab-run

**Pull request merged**